src/mobile-token/use-token-toggle-details.ts

```diff
@@ -1,36 +1,43 @@
-import {useEffect, useState} from 'react';
 import {useMobileTokenContextState} from '@atb/mobile-token';
+import { useIsFocused } from '@react-navigation/native';
+import {useQuery} from '@tanstack/react-query';
 
-const useTokenToggleDetails = (shouldFetchTokenDetails: boolean) => {
-  const [shouldShowLoader, setShouldShowLoader] = useState<boolean>(false);
-  const [toggleLimit, setToggleLimit] = useState<number | undefined>();
-  const [maxToggleLimit, setMaxToggleLimit] = useState<number | undefined>();
+/**
+ * 10 seconds stale/cache time to allow user get latest result in case they switch their phone/t:card
+ */
+const ONE_MINUTE = 1000 * 60;
+export const GET_TOKEN_TOGGLE_DETAILS_QUERY_KEY = 'getTokenToggleDetails';
 
-  const {getTokenToggleDetails} = useMobileTokenContextState();
+const useTokenToggleDetails = () => {
+  const {getTokenToggleDetails, deviceInspectionStatus} = useMobileTokenContextState();
+  const screenHasFocus = useIsFocused();
+  const shouldFetchTokenDetails =
+    screenHasFocus && deviceInspectionStatus !== 'loading';
+  return useQuery({
```

Comment:
It could be considered to do like this:
```
const queryResult = useQuery({...});
return {
    ...queryResult,
    toggleLimit: queryResult.data?.toggleLimit,
    maxToggleLimit: queryResult.data?.maxToggleLimit,
  }
```
Then the usage of the hook is simplified:
`const {toggleLimit, maxToggleLimit, isLoading} = useTokenToggleDetails();`